HW2/P2/P2.txt

```diff
@@ -0,0 +1,29 @@
+The serial approach performed much better than the parallel approach in both cases. This makes sense because the overhead of using the locks takes more time than simply moving the data.
+
+In the uncorrelated case the fine-grained locks performed better than any of coarse grained ones. In other words, no matter how few pieces of data were assigned to a lock, having less locks always reduced performance. This is a reasonable result because having more data per lock (a higher N) leads to more blocking. With uncorrelated data it makes sense to make N as small as possible (1)
+
+However, in the correlated case, the coarse-grained strategy can outperform the fine-grained. This is most liekly the result of many of the transactions only needing one lock (due to the correlation). When a transaction only needs one lock, there is less overhead but the chance of blocking does not substantially increase. The empirical results show that it's ideal to have ~20 pieces of data for each lock (N=20). 
+
```

Comment:
~20 sounds good! I think it would be good to note the theoretical reason for this result.